governance.md

```diff
@@ -0,0 +1,164 @@
+
+It also has a roadmap for additions coming in the next iteration of governance (Governane 1.1).
+
+## Op1st Community Governance 1.0 (MVG)
+
+A healthy open source project is one that demonstrates open practices, uses open infrastructure, and cultivates an open culture with the goal of becoming more sustainable.
+
+### Principles
+As a new cloud-native community growing up at the footsteps of massively scaled and successful projects such as Kubernetes and OpenInfra, we want to be careful to learn from those who came before us, while forging our own path into the challenges we've chosen to undertake.
+
+To stand up this Governance for a 1.0 good-enough minimal-viable-governance (MVG), we are looking to our key predecessors for some of our starting points.
+As an initial set of principles we are combining elements of the [Kubernetes community principles](https://github.com/kubernetes/community/blob/master/governance.md#principles) and the [OpenInfra Foundations Four Opens](https://openinfra.dev/four-opens/).
+
+#### Op1st Community Principles 1.0
+Our mission requires us to use open source principles in ways beyond licensing the code we write, but at the core we are an open source community and want to adhere to the best possible principles for creating an open, welcoming, and inclusive community.
+
+These are the Op1st community principles we adhere to:
+1. **Open Source** : Any content we create is released under an open source license. Content includes software, documentations, configuration files, and training materials.
+  - This is crucial to fulfilling our mission around open sourcing operations
+2. **Welcoming and respectful** : We intend to be an open, welcoming, and inclusive community.
+Diversity is our strength.
+We have and follow a Code of Conduct.
+  - We currently follow the OpenInfra Foundation Code of Conduct as a member of the OpenInfra Labs.
+3. **Open Design** : We follow a transparent and open process for planning and designing any content, services, and features of Op1st.
+Ideas and contributions are accepted according to their technical merit and alignment with project objectives, scope, and design principles.
+4. **Open Work** : Two of our core tools for successful community building are transparency and accessibility.
+To support that, work and collaboration must be done in public to the greatest degree possible.
+Refer to the section on Community Groups and SIGs in particular for details on how this is performed.
+5. **Truly Open Community** : The best way to build a sustainable open source community is to ensure the community is a cohesive, inclusive, and equitable place where all the voices are heard and anyone can rise to leadership positions.
+
+### Code of Conduct
+
+We are using the [upstream OpenInfra Foundation Code of Conduct](https://openinfra.dev/legal/code-of-conduct) until we can convene the new Community SIG to lead an open effort to create our own, modern code of conduct.
```

Comment:
~~~modern~~~ this implies theirs is old and outdated, which I hope it's not 😄

Reply:
Definitely not "outdated" as the board of directors just voted to approve updates to it a few months ago.

More to the point, the idea is for that Community Code of Conduct to be usable by all projects represented by the Open Infrastructure Foundation. If you have feedback on how to improve it to be more useful for Operate First, then that would be entirely welcome. Probably the most fruitful avenues for amending it would be to start a discussion on the foundation mailing list, or join the Diversity and Inclusion Working Group meetings as they have a fair amount of experience with the current state of the document and process of engaging with the foundation's staff, board, and legal counsel in order to drive improvements there or in similar policy documents.

Reply:
I don't think the "until" clause should be listed here. Is there a reason to say that "things may change in future" in here? I think that's actually implied in all sections, isn't it?

```suggestion
We are using the [upstream OpenInfra Foundation Code of Conduct](https://openinfra.dev/legal/code-of-conduct).
```

Reply:
Should there be a mention that individual projects that are deployed on Op1st (e.g. Strimzi / Kafka) may have their own CoC, which would apply appropriately?

Reply:
@durandom @fungi @tumido You raise good questions, let me break down what is happening in this sentence; we can come to conclusion on what it should be as a good-enough to start.

1. This community never chose<sup>[1](#fn01)</sup> its code of conduct, because this community has not previously had a method for making a community-wide decision in an open source way. One goal of this section is to make it clear that we need to have the actual step of this community discussing and deciding on a code of conduct. It may be as simple as agreeing to continue using the OI Foundation Code of Conduct.
2. _However_, there is activity right now in the ecosystem looking at the process of adopting codes of conduct, the templates they are modeled from, and how projects administer and enforce them. We would be remiss to skip the opportunity to look at things with the freshest perspectives available.
3. Finally, to @pilhuhn's point, we need to thing of our CoC in light of also being a platform for the success of other projects, and we need to consider that through so we remain aligned.

@tumido I agree fully about all things being up for change in the 1.1 version. My reason for saying this in particular among all others is for those who are looking beyond the existence of a Code of Conduct for a community actively practicing inclusion. Currently, that means demonstrating an awareness of the above points and where we are at this moment in history with regards to practices and discussion around codes of conduct. Which basically is, "Currently our practices are under discussion, it's worth taking the extra time to talk it through especially for a new community."

@fungi thank you, those offers are much appreciated and give us some good ways to setup a transparent and equitable process for this community. The Community SIG will drive this discussion (and I specifically invite you to participate in that SIG, at least if you are able to for this first year as we work out many initialisms), the discussion might look like this:

1. We put out a call for participation in a CoC working group, giving a few weeks to form and have the first meeting so we can get the most participation.
    - We MUST have an equitable asynchronous process for participation, including fairly long time arcs, regular reminders of status, zero requirement for real-time meetings (or perhaps requiring zero real-time meetings?), and so forth.
2. Once the CoC WG formally engages, some level of working together and bonding is necessary to be able to engage with the OpenInfra Foundation processes.
3. When ready, the WG engages with the OI Foundation D&I WG to begin a working relationship and see where things are at that point.
    - In my mind, it would be great if we could surface the majority of discussion points and change requests from the community via these first few interactions. Then anything in the future has a change management process it can follow and typically a longer lead time.
4. If any changes or other items are needed, the two WGs collaborate on a presentation or proposal to the OI Foundation Board.

How about these adjustments?
```suggestion
We are using the [upstream OpenInfra Foundation Code of Conduct](https://openinfra.dev/legal/code-of-conduct)(CoC).
In creating the next Governance 1.1 we will have an actual process to make decisions as a community. We can then either formally ratify this same CoC, or the Community SIG can convene a cross-project working group to discuss and decide an inclusive and equitable process for a new code of conduct.
```

<a name="fn01">1</a>: I couldn't find a CoC when I did my initial audit of this community. It not only wasn't linked anywhere, it wasn't clear there even was one. That's a Sev1 (high priority thing to fix) from a community architect point of view, so I paused my audit to file my first PR for this project: https://github.com/operate-first/operate-first.github.io/pull/244 from 9c2d0e6.

Reply:
I went looking on both the OpenInfra Labs and Operate First websites and they both point to the OIF CoC. I'm not sure if you added the link on the Operate First site after this document was created @quaid or if it was there already there though. @fungi might know the process more of how projects are set up within OIF as to whether utilizing the OIF CoC is part of the agreement. 

As far as working with the OIF D&I WG, we are always inviting all the projects to the meetings, I'm actually not sure if any of the sub-projects have their own D&I groups or not. The WG is available to all projects to help with D&I issues from helping with Inclusive Naming, D&I Surveys, etc. and reports back to the Foundation Board. While originally it was just OpenStack as the only project at the time, it never was actually part of OpenStack if that helps clarify anything.

Reply:
Amy's comment just reminded me... when I replied by E-mail a few weeks ago, GitHub mis-threaded it into general comments instead of into the file comments, so I'm copying it here for better context. Sorry!

On 2021-12-18 16:05:13 -0800 (-0800), Karsten Wade wrote:
[...]
> 1. This community never chose<sup>[1](#fn01)</sup> its code of
> conduct, because this community has not previously had a method
> for making a community-wide decision in an open source way. One
> goal of this section is to make it clear that we need to have the
> actual step of this community discussing and deciding on a code of
> conduct. It may be as simple as agreeing to continue using the OI
> Foundation Code of Conduct.

I suppose it's all a matter of perspective. The Open Infrastructure
community chose a code of conduct and continues to maintain it with
the help of its foundation, members, and board of directors. The
OpenInfra Labs project chose to adopt that code of conduct and help
shape any future revisions thereof by joining the foundation which
represents the Open Infrastructure community. Op1st, as a subproject
of the OpenInfra Labs project (or "key development activity" in the
words of the project's website), is covered by that project's
commitment to follow the Open Infrastructure Community Code of
Conduct.

> 2. _However_, there is activity right now in the ecosystem looking
> at the process of adopting codes of conduct, the templates they
> are modeled from, and how projects administer and enforce them. We
> would be remiss to skip the opportunity to look at things with the
> freshest perspectives available.
[...]

I completely concur, this is an excellent opportunity to revisit the
intent and wording of the Community Code of Conduct and help revise
it to the benefit of the Open Infrastructure Community, of which
Op1st is a part.

> @fungi thank you, those offers are much appreciated and give us
> some good ways to setup a transparent and equitable process for
> this community. The Community SIG will drive this discussion (and
> I specifically invite you to participate in that SIG, at least if
> you are able to for this first year as we work out many
> initialisms), the discussion might look like this:
[...snip proposed plan...]

This sounds like a reasonable approach to me, and I'm happy to
(continue to) join in these discussions. Just to underscore my
perspective, Op1st does already have a code of conduct which it
shares with a community of projects. Much like open source software,
open communities are easy to fork but such forks should be taken
with care, and ideally start with attempts to collaborate in the
maintenance of the resources shared with the rest of the community.

> <a name="fn01">1</a>: I couldn't find a CoC when I did my initial
> audit of this community. It not only wasn't linked anywhere, it
> wasn't clear there even was one. That's a Sev1 (high priority
> thing to fix) from a community architect point of view, so I
> paused my audit to file my first PR for this project:
> https://github.com/operate-first/operate-first.github.io/pull/244
> from 9c2d0e6.

To be clear, the CoC has been declared at the bottom of each page of
the project Web site since
https://github.com/open-infrastructure-labs/openinfralabs-website/commit/78dd7e1
merged in August of 2020. I agree it's not a bad idea for
subprojects to mention the CoC in their documentation as well, but
there seems to be a lot of fragmentation across OpenInfra Labs at
present, pointing to a need to establish a clear project governance
while the project itself works toward increased cohesion (made more
evident by one of its subprojects drafting its own formal governance
even before the project has initiated such a discussion).

Reply:
> I went looking on both the OpenInfra Labs and Operate First websites and they both point to the OIF CoC. I'm not sure if you added the link on the Operate First site after this document was created @quaid or if it was there already there though. @fungi might know the process more of how projects are set up within OIF as to whether utilizing the OIF CoC is part of the agreement.

Other way around, I had us add the OIF CoC in May 2021. At that time no one in the project was aware of any process or for that matter seemed to consider themselves to be covered by any formal code of conduct. This was in operate-first/operate-first.github.io@9c2d0e6 .
 
> As far as working with the OIF D&I WG, we are always inviting all the projects to the meetings, I'm actually not sure if any of the sub-projects have their own D&I groups or not. The WG is available to all projects to help with D&I issues from helping with Inclusive Naming, D&I Surveys, etc. and reports back to the Foundation Board. While originally it was just OpenStack as the only project at the time, it never was actually part of OpenStack if that helps clarify anything.

Thanks, that helps. Now that we have are going to have a governance, we can actually have a way to interact with the OIF, including on DEI matters.

Reply:
> I suppose it's all a matter of perspective. The Open Infrastructure community chose a code of conduct and continues to maintain it with the help of its foundation, members, and board of directors. The OpenInfra Labs project chose to adopt that code of conduct and help shape any future revisions thereof by joining the foundation which represents the Open Infrastructure community. Op1st, as a subproject of the OpenInfra Labs project (or "key development activity" in the words of the project's website), is covered by that project's commitment to follow the Open Infrastructure Community Code of Conduct.

That is the logic I followed in opening operate-first/operate-first.github.io#244—I wouldn't have pushed an active community to accept being under a code of conduct without any kind of process to get there, but it was also reasonable to conclude that by dint of relationship we were already under the OIF CoC.

It has been since May 2021 that the conversations about CoCs have grown, so that was part of the reasoning here. The other part is that no matter what, it's a good step for the Op1st community to formally recognize their CoC. E.g. as part of the discussion we'll want to have some training materials, resources, reporting pathways, etc. so community members have a stronger understanding.
 
> To be clear, the CoC has been declared at the bottom of each page of the project Web site since [open-infrastructure-labs/openinfralabs-website@78dd7e1](https://github.com/open-infrastructure-labs/openinfralabs-website/commit/78dd7e1) merged in August of 2020. I agree it's not a bad idea for subprojects to mention the CoC in their documentation as well, but there seems to be a lot of fragmentation across OpenInfra Labs at present, pointing to a need to establish a clear project governance while the project itself works toward increased cohesion (made more evident by one of its subprojects drafting its own formal governance even before the project has initiated such a discussion).

On the former point, I think it's crucial and it's recommended practice to better highlight Codes of Conduct. In fact, I'm not happy with where it is right now and have Community SIG issue operate-first/community#54 about that.

On the latter point, I honestly considered going to OI Labs first to get governance there first so we could be given the charter to create an Op1st governance. I think wisely I'm keeping within where I can help make things happen, but I would he happy and interested in helping OI Labs put a governance together.

(I'm always finding it important in conversations for people to understand the relationship between the conceptual work happening in OI Labs and the prototype work happening in Op1st. Getting governance pieces to fit as parts of the puzzle will help, especially if it's a nice diagram on the OI Labs site.)

Reply:
OK, the way I worded it in the suggestion above made it into one of these commits :) and it seems we're in agreement to proceed and keep discussing these topics in their new home of the Community SIG.

I'm resolving this to make it clear there no more actions needed for this PR, and we can come later for seeding future Gov 1.1 discussions.